src/kboot.c

```diff
@@ -207,7 +207,8 @@ static int dt_set_chosen(void)
         printf("FDT: initrd at %p size 0x%lx\n", initrd_start, initrd_size);
     }
 
-    if (cur_boot_args.video.base) {
+    if (cur_boot_args.video.base &&
+        (adt_path_offset(adt, "/arm-io/disp0") > 0 || (chip_id != T8012 && chip_id != T7000))) {
```

Comment:
Could you also add a comment about this condition? Looks somewhat confusing to me

Reply:
Sure, this condition is that there exist T8012 and T7000 devices with no display at all.

Reply:
Okay, but do they still have a /arm-io/disp0 node then?

Reply:
That node doesn't exist in their ADT.

Reply:
okay, so please take a step back and explain what you are trying to do here because this condition doesn't make much sense to me.

Right now we will init the framebuffer if
cur_boot_args.video.base is not null and either
    /arm-io/disp0 exists
        OR
    the chip is not T8012 and also not T7000

Reply:
This condition is to ensure on chips that definitely have a display attached, the framebuffer will be initialized if passed by iBoot, but now I am aware that some devices uses `/arm-io/dispext4` and this would actually break that, so how about:

Init framebuffer if base exists and (either chip is S5L8960X or the dimensions are not 640x1136)
(Would miss 640x1136 external display (?) does those exist (?))

Reply:
Does it even hurt to initialize the framebuffer when no display is connected?

Reply:
So why not just check if cur_boot_args.video.base is set and /chosen/framebuffer exists then?